core/src/main/java/uk/ac/bbsrc/tgac/miso/core/data/QC.java

```diff
@@ -23,66 +23,85 @@
 
 package uk.ac.bbsrc.tgac.miso.core.data;
 
+import java.io.Serializable;
 import java.util.Date;
 
-import com.eaglegenomics.simlims.core.Securable;
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+import javax.persistence.JoinColumn;
+import javax.persistence.ManyToOne;
+import javax.persistence.MappedSuperclass;
+import javax.persistence.Temporal;
+import javax.persistence.TemporalType;
 
+import com.eaglegenomics.simlims.core.User;
+
+import uk.ac.bbsrc.tgac.miso.core.data.impl.UserImpl;
 import uk.ac.bbsrc.tgac.miso.core.data.type.QcType;
 
-/**
- * A QC represents a validation step carried out on a given model object, e.g. a {@link Library} via a {@link LibraryQC}, a {@link Sample}
- * via a {@link SampleQC}, or a {@link Run} via a {@link RunQC}
- * 
- * @author Rob Davey
- * @since 0.0.2
- */
-public interface QC extends Securable, Comparable<QC>, Deletable {
-  public long getId();
-
-  public void setId(long id);
-
-  /**
-   * Returns the qcCreator of this QC object.
-   * 
-   * @return String qcCreator.
-   */
-  public String getQcCreator();
-
-  /**
-   * Sets the qcCreator of this QC object.
-   * 
-   * @param creator
-   *          qcCreator.
-   */
-  public void setQcCreator(String creator);
-
-  /**
-   * Returns the qcMethod of this QC object.
-   * 
-   * @return String qcMethod.
-   */
-  public QcType getQcType();
-
-  /**
-   * Sets the QcType of this QC object.
-   * 
-   * @param type
-   *          type.
-   */
-  public void setQcType(QcType type);
-
-  /**
-   * Returns the qcDate of this QC object.
-   * 
-   * @return Date qcDate.
-   */
-  public Date getQcDate();
-
-  /**
-   * Sets the qcDate of this QC object.
-   * 
-   * @param date
-   *          qcDate.
-   */
-  public void setQcDate(Date date);
+@MappedSuperclass
+public abstract class QC implements Serializable {
+  private static final long serialVersionUID = 1L;
+
+  public static final Long UNSAVED_ID = 0L;
+
+  @ManyToOne(targetEntity = UserImpl.class)
+  @JoinColumn(name = "creator")
+  private User creator;
+
+  @Temporal(TemporalType.DATE)
+  private Date date = new Date();
+
+  @Id
+  @GeneratedValue(strategy = GenerationType.AUTO)
+  private long qcId = QC.UNSAVED_ID;
+
+  private Double results;
+
+  @ManyToOne
+  @JoinColumn(name = "type")
+  private QcType type;
+
+  public User getCreator() {
+    return creator;
+  }
+
+  public Date getDate() {
+    return date;
+  }
+
+  public abstract QualityControllable<?> getEntity();
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Remove usage of generic wildcard type. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1452)